android/libraries/rib-test/src/main/kotlin/com/uber/rib/core/RouterHelper.kt

```diff
@@ -0,0 +1,158 @@
+   */
+  @JvmStatic
+  open fun detach(router: Router<*>) {
+    router.dispatchDetach()
+  }
+
+  /**
+   * Verifies that the [Router] is attached.
+   *
+   * @param router the [Router].
+   */
+  @JvmStatic
+  open fun verifyAttached(router: Router<*>) {
+    verify(router).dispatchAttach(or(isNull(), isA(Bundle::class.java)), anyString())
+  }
+
+  /**
+   * Verifies that the [Router] was attached a certain number of times
+   *
+   * @param router the [Router].
+   * @param times the number of times attached
+   */
+  @JvmStatic
+  open fun verifyAttached(router: Router<*>, times: Int) {
+    verify(router, times(times)).dispatchAttach(or(isNull(), isA(Bundle::class.java)), anyString())
+  }
+
+  /**
+   * Verifies that the [Router] is attached.
+   *
+   * @param order [InOrder] for ordered verification.
+   * @param router the [Router].
+   */
+  @JvmStatic
+  open fun verifyAttached(order: InOrder, router: Router<*>) {
+    order.verify(router).dispatchAttach(or(isNull(Bundle::class.java), isA(Bundle::class.java)), anyString())
+  }
+
+  /**
+   * Verified that the [Router] is attached with a sepcific tag.
+   *
+   * @param router the [Router].
+   * @param tag the expected tag.
+   */
+  @JvmStatic
+  open fun verifyAttached(router: Router<*>, tag: String) {
+    verify(router).dispatchAttach(or(isNull(Bundle::class.java), isA(Bundle::class.java)), eq(tag))
+  }
+
+  /**
+   * Verifies that the [Router] is attached with an additional [VerificationMode].
+   *
+   * @param router the [Router].
+   * @param mode The mockito verification mode. ie. `times(1)`.
+   */
+  @JvmStatic
+  open fun verifyAttached(router: Router<*>, mode: VerificationMode) {
+    verify(router, mode).dispatchAttach(or(isNull(Bundle::class.java), isA(Bundle::class.java)), anyString())
+  }
+
+  /**
+   * Verifies that the [Router] is not attached.
+   *
+   * @param router the [Router].
+   */
+  @JvmStatic
+  open fun verifyNotAttached(router: Router<*>) {
+    verify(router, never())
+      .dispatchAttach(or(isNull(Bundle::class.java), isA(Bundle::class.java)), anyString())
+  }
+
+  /**
+   * Verifies that the [Router] is detached.
+   *
+   * @param router the [Router].
+   */
+  @JvmStatic
+  open fun verifyDetached(router: Router<*>) {
+    verify(router).dispatchDetach()
+  }
+
+  /**
+   * Verifies that the [Router] is detached.
+   *
+   * @param order [InOrder] for ordered verification.
+   * @param router the [Router].
+   */
+  @JvmStatic
+  open fun verifyDetached(order: InOrder, router: Router<*>) {
+    order.verify(router).dispatchDetach()
+  }
+
+  /**
+   * Verifies that the [Router] is detached with an additional [VerificationMode].
+   *
+   * @param router the [Router].
+   * @param mode The mockito verification mode. ie. `times(1)`.
+   */
+  @JvmStatic
+  open fun verifyDetached(router: Router<*>, mode: VerificationMode) {
+    verify(router, mode).dispatchDetach()
+  }
+
+  /**
+   * Verifies that the [Router] is detached.
```

Comment:
nit - Should this be verify is _not_ detached?

Reply:
c2057701daf5c855b12768f8765c39b0ef2f0b13